.github/workflows/deploy-web-app-ec2.yml

```diff
@@ -0,0 +1,27 @@
+name: Deploy on aws EC2
+on:
+  push:
+    branches:
+    - main
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    environment: Staging AWS
+    
+    steps:
+      - uses: actions/checkout@v2 
+
+      - name: Deploy in EC2
+        env:
+            PRIVATE_KEY: ${{ secrets.AWS_EC2_PRIVATE_KEY  }}
+            HOSTNAME : ${{ secrets.AWS_EC2_PUBLIC_IP  }}
+            USER_NAME : ${{ secrets.AWS_EC2_USERNAME  }}
+            
+        run: |
+          echo "$PRIVATE_KEY" > private_key && chmod 600 private_key
+          rsync --delete -r -e "ssh -o StrictHostKeyChecking=no -i private_key" --exclude="mobile-app/*" ./ ${USER_NAME}@${HOSTNAME}:~/Hex
```

Comment:
Add a --filter param to use gitignore, when you will run test on CI, you will have to install dependencies and the node_modules folder will be rsynced by this step.